common/src/WooCommerce.php

```diff
@@ -143,6 +133,32 @@ public function order_completed($order_id) {
         }
     }
 
+    public function addGtinOption() {
+        $gtin_handler = new GtinHandler(null, $this->plugin);
```

Comment:
No need to change this if you don't feel like it now, but:

The fact that you're using `null` here for the product should tell you that this doesn't really belong in the constructor, but rather should be passed as a method argument whenever it's needed.  The way you do it now makes certain methods of the GtinHandler "unsafe", as they will try to call methods on the product which is not set.

Class constructor arguments should be "dependencies", ie, other classes that it needs to function.  The product is more like a value that is used in a particular call.  This way you can initialize the GtinHandler once, and call the `getGtin` method for each product.